tests/CookieTest.php

```diff
@@ -19,62 +20,79 @@
  */
 final class CookieTest extends TestCase
 {
-    public function testDefaults(): void
+	/** @dataProvider \Ergebnis\DataProvider\StringProvider::arbitrary() */
+    public function testDefaults(string $name): void
     {
-        $newCookie = new marcoschwepp\Cookie\Cookie('testCookie');
+        $newCookie = new marcoschwepp\Cookie\Cookie($name);
 
-        self::assertSame($newCookie->getName(), 'testCookie');
+        self::assertSame($newCookie->getName(), $name);
         self::assertSame($newCookie->getPath(), '/');
-
+		self::assertSame($newCookie->getDomain(), '');
         self::assertFalse($newCookie->isSecure());
         self::assertFalse($newCookie->isHttpOnly());
-
         self::assertEmpty($newCookie->getValue());
-
         self::assertInstanceOf(\DateTimeImmutable::class, $newCookie->getExpiresAt());
     }
 
-    public function testConstructFromOptions(): void
+	/** @dataProvider \Ergebnis\DataProvider\StringProvider::arbitrary() */
+    public function testConstructFromOptions(string $name): void
     {
         $options = [
-            'name' => 'testCookie',
+            'name' => $name,
         ];
 
         $newCookieFromOptions = marcoschwepp\Cookie\Cookie::constructFromOptions($options);
 
-        self::assertSame($newCookieFromOptions->getName(), 'testCookie');
+        self::assertSame($newCookieFromOptions->getName(), $name);
         self::assertSame($newCookieFromOptions->getPath(), '/');
-
         self::assertFalse($newCookieFromOptions->isSecure());
+		self::assertSame($newCookieFromOptions->getDomain(), null);
         self::assertFalse($newCookieFromOptions->isHttpOnly());
-
         self::assertEmpty($newCookieFromOptions->getValue());
-
         self::assertInstanceOf(\DateTimeImmutable::class, $newCookieFromOptions->getExpiresAt());
     }
 
-    public function testGettersAndSetters(): void
+	/** @dataProvider \DataProvider::cookie() */
+    public function testGettersAndSetters(
```

Comment:
I wrapped up this completely for you. Your old test was not covering many.
With DataProviders it should be better.